src/sentry/integrations/slack/utils.py

```diff
@@ -378,6 +380,33 @@ def get_channel_id(organization, integration, name, use_async_lookup=False):
     return get_channel_id_with_timeout(integration, name, timeout)
 
 
+def validate_channel_id(name: str, integration_id: int, input_channel_id: str) -> None:
+    """
+    In the case that the user is creating an alert via the API and providing the channel ID and name
+    themselves, we want to make sure both values are correct.
+    """
+    try:
+        integration = Integration.objects.get(id=integration_id)
+    except Integration.DoesNotExist:
+        raise Http404
+
+    headers = {"Authorization": "Bearer %s" % integration.metadata["access_token"]}
+    payload = {"channel": input_channel_id}
+    client = SlackClient()
+    try:
+        results = client.get("/conversations.info", headers=headers, params=payload)
+    except ApiError as e:
+        if e.text == "channel_not_found":
+            raise ValidationError("Channel not found. Invalid ID provided.")
+        logger.info("rule.slack.conversation_info_failed", extra={"error": str(e)})
+        raise IntegrationError("Could not retrieve Slack channel information.")
+    if not (
+        strip_channel_name(name) == results["channel"]["name"]
+        and input_channel_id == results["channel"]["id"]
```

Comment:
Probably don't need to validate this, since we are already fetching it by channel id?